File: go/stats.go
package main

import "fmt"

func Sum(values []float64) float64 {
	var total float64

	for _, v := range values {
		total += v
	}

	return total
}

func Mean(values []float64) (float64, error) {
	if len(values) == 0 {
		return 0, fmt.Errorf("Mean of empty slice")
	}

	return Sum(values) / float64(len(values)), nil
}

File: python/stats.py
class StatsError(Exception):
    pass


def mean(values: list) -> int:
    if not values:
        raise StatsError('"mean" with no values')
    return sum(values) / len(values)


if __name__ == '__main__':
    values = [2, 4, 8]
    try:
        print(mean(values))
    except StatsError as error:
        print(f'error: {error}')
